### PR TITLE
Modernize the CSS of the U2 MD views

### PR DESCRIPTION
**Pull request by shepheb**
Now that the shorthand is checked in, use it on the various MD U2 views.

Also adds the halos to several that were missing it.

### CODE REVIEW

File: core/mm4Method.js [hunk 1]
```diff
@@ -255,7 +255,7 @@ CLASS({
     function maybeCall(X, that) { /* Executes this action if $$DOC{ref:'.isEnabled'} is allows it. */
       if ( this.isAvailable.call(that, this) && this.isEnabled.call(that, this) ) {
         this.code.call(that, X, this);
-        that.publish(['action', this.name], this);
+        if (that && that.publish) that.publish(['action', this.name], this);
```
Comment (kgrgreer) [inline core/mm4Method.js]: When does this happen?

File: js/foam/u2/md/ActionButton.js [hunk 2]
```diff
@@ -64,19 +82,37 @@ CLASS({
 
   methods: [
     function initE() {
-      return this.cls('md-button').style({
-        color: this.color_$,
-        opacity: this.alpha$,
-      })
-        .on('click', this.onClick)
+      var typeClasses = {
```
Comment (kgrgreer) [inline js/foam/u2/md/ActionButton.js]: Should this be a constant?

File: core/mm4Method.js (hunk 1 above)
Comment (shepheb) [inline core/mm4Method.js]: When called from `js/foam/u2/md/ActionButton.js` line 132, it passes `this.data` as `that`. For a view with no `data` that does `this.add(this.MY_ACTION)` for an action of its own, `this.data` is blank. Seems like a legit case to me.

Or maybe the `data` of the `ActionButton` should always be the instance that owns the `Action`?

File: js/foam/u2/md/ActionButton.js (hunk 2 above)
Comment (shepheb) [inline js/foam/u2/md/ActionButton.js]: Definitely. Done.

File: core/mm4Method.js (hunk 1 above)
Comment (kgrgreer) [inline core/mm4Method.js]: I think it would make more sense for 'that' to be the Action owner.

**Comment by shepheb** [comment]
Okay, the controversial stuff is gone. Merging this in.

**Pull request closed without merging**